test/test_utils/security.py

```diff
@@ -63,6 +64,53 @@ def construct_allowlist_from_ecr_scan_result(self, vulnerability_list):
             if CVESeverity[vulnerability["severity"]] >= self.minimum_severity:
                 self.vulnerability_list[package_name].append(vulnerability)
         return self.vulnerability_list
+    
+    def get_flattened_vulnerability_list(self):
+        """
+        Returns the vulnerability list in the flattened format. For eg., if a vulnerability list looks like
+        {"k1":[{"a":"b"},{"c":"d"}], "k2":[{"e":"f"},{"g":"h"}]}, it would return the following:
+        [{"a":"b"},{"c":"d"},{"e":"f"},{"g":"h"}]
+
+        :return: List(dict)
+        """
+        if self.vulnerability_list:
+            return [
+                vulnerability
+                for package_vulnerabilities in self.vulnerability_list.values()
+                for vulnerability in package_vulnerabilities
+            ]
+        return None
+
+    def sort_dictionary_in_custom_way(self, input_dict):
+        """
+        This method is specifically made to sort the vulnerability list which is actually a dict 
+        and has the following structure:
+        {
+            "packge_name1":[
+                {"name":"cve-id1", "uri":"http.." ..},
+                {"name":"cve-id2", "uri":"http.." ..}
+            ],
+            "packge_name2":[
+                {"name":"cve-id1", "uri":"http.." ..},
+                {"name":"cve-id2", "uri":"http.." ..}
+            ]
+        }
+        We want to first sort the innermost list of dicts based on the "name" of each dict and then we sort the
+        otermost dict based on keys i.e. package_name1 and package_name2
+        :param input_dict: dict(key, list(dict)), represents vulnerability_list
+        :return: dict, input_dict sorted in a custom way
+        """
+        copy_dict = copy.deepcopy(input_dict)
+        for _,list_of_dict in copy_dict.items():
+            list_of_dict.sort(key=lambda dict_element:dict_element["name"])
+        od = collections.OrderedDict(sorted(copy_dict.items()))
+        return dict(od)
```

Comment:
Not sure I understand this. You sort `list_of_dict` and think that's enough. 
Then you sort copy_dict but convert it to `OrderedDict` to preserve the sorted order.
But after that you convert `OrderedDict` into a regular dict. 

So I think you should either don't use OrderedDict at all or convert everything into OrderedDict.

Reply:
Great suggestion, made the changes accordingly. Thanks!!